cpp-client/deephaven/examples/demos/feedtimes.cc

```diff
@@ -0,0 +1,312 @@
+        if (!initial) {
+          std::cout << ", max_delay=" << FormatMicros(max_delay);
+        }
+      } catch (const std::runtime_error &ex) {
+        std::cout << "invalid_time(" << max.Nanos() << ")";
+      }
+      std::cout << "."
+                // we actually want the flush, so we use endl instead of '\n'.
+                << std::endl;
+    }
+    last_time = now;
+  }
+
+  void ProcessDeltas(
+      const my_time_point recv_ts,
+      const ClientTable &table,
+      const size_t col_index,
+      std::shared_ptr<RowSequence> rows) {
+    const auto generic_datetime_col = table.GetColumn(col_index);
+    const auto *typed_datetime_col =
+        VerboseCast<const DateTimeColumnSource*>(DEEPHAVEN_LOCATION_EXPR(generic_datetime_col.get()));
+
+    constexpr const size_t kChunkSize = 8192;
+    auto data_chunk = DateTimeChunk::Create(kChunkSize);
+
+    while (!rows->Empty()) {
+      auto these_rows = rows->Take(kChunkSize);
+      auto these_rows_size = these_rows->Size();
+      nrows_ += these_rows_size;
+      rows = rows->Drop(kChunkSize);
+
+      typed_datetime_col->FillChunk(*these_rows, &data_chunk, nullptr);
+
+      auto data = data_chunk.data();
+
+      DateTime v;
+      size_t j;;
+      // Find first non_null value
+      for (j = 0; j < these_rows_size; ++j) {
+        v = data[j];
+        if (!DateTime::isNull(v)) {
+          break;
+        }
+      }
+
+      if (j >= these_rows_size) {
+        continue;
+      }
+
+      DateTime lmin = v;
+      DateTime lmax = v;
+
+      for (size_t i = j; i < these_rows_size; ++i) {
+        v = data[i];
+        if (DateTime::isNull(v)) {
+          continue;
+        }
+        if (v < lmin) {
+          lmin = v;
+        } else if (v > lmax) {
+          lmax = v;
+        }
+      }
+
+      if (lmin < min_) {
+        min_ = lmin;
+        min_delay_ = recv_ts - ToTimePoint(lmin);
+      }
+      if (lmax > max_) {
+        max_ = lmax;
+        max_delay_ = recv_ts - ToTimePoint(lmax);
+      }
+    }
+  }
+
+private:
+  void Reset() {
+    min_ = DateTime::Max();
+    max_ = DateTime::Min();
+    min_delay_ = my_duration::max();
+    max_delay_ = my_duration::min();
+    nrows_ = 0UL;
+    nupdates_ = 0UL;
+  }
+
+  void ReadAndReset(bool &initial,
+                    DateTime &min, DateTime &max, my_duration &min_delay, my_duration &max_delay,
+                    uint64_t &nrows, uint64_t &nupdates) {
+    std::unique_lock lock(mux_);
+    initial = update_idx_ == 1UL;
+    min = min_;
+    max = max_;
+    min_delay = min_delay_;
+    max_delay = max_delay_;
+    nrows = nrows_;
+    nupdates = nupdates_;
+    Reset();
+  }
+
+private:
+  mutable std::mutex mux_;
+  mutable std::condition_variable cond_;
+  const char *col_name_;
+  DateTime min_;
+  DateTime max_;
+  my_duration min_delay_;
+  my_duration max_delay_;
+  std::uint64_t update_idx_;
+  std::uint64_t nrows_;
+  std::uint64_t nupdates_;
+  bool initial_done_;
```

Comment:
As a coding style matter, I would recommend all these be set to something here:

```
  std::uint64_t update_idx_ = 0;
  std::uint64_t nrows_ = 0;
  std::uint64_t nupdates_ = 0;
  bool initial_done_ = false;
```

... even though it's redundant. We once had a conversation about this. It's a belt-and-suspenders protection against an uninitialized variable having a random value and being hard to debug. (As we concluded, deterministic values -- even if wrong -- are much easier to debug than nondeterministic values)